### PR TITLE
Enhance/readme

### PR DESCRIPTION
**Pull request by brianignacio5**
README is too long. Separating into smaller docs files and adding links to README.

### CODE REVIEW

File: docs/DEBUGGING.md [hunk 1]
```diff
@@ -17,10 +19,24 @@ Default values launch.json for ESP-IDF Debug Adapter:
...
+- `appOffset`: Program start address offset to start debugging.
 - `debugPort`: Port for ESP-IDF Debug Adapter. Default: 43474.
+- `env`: Environment variables to apply to the ESP-IDF Debug Adapter.
+- `gdbinitFile`: Specify the gdbinit file to send to gdb.
+- `initGdbCommands`: One or more xtensa-esp32-elf-gdb commands to execute in order to setup the underlying debugger. If gdbinitFile is defined, these commands will be ignored.
 - `logLevel`: Debug Adapter Debug level (0-4), 5 - for a full OOCD log. Default: 2.
 - `mode`: Can be either `auto`, to start the debug adapter and openOCD server within the extension or `manual`, to connect to existing debug adapter and openOCD session.
-- `env`: Environment variables to apply to the ESP-IDF Debug Adapter.
+
+> **NOTE:** As shown in [JTAG Debugging debugging with command line](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/jtag-debugging/using-debugger.html#command-line) make sure to include the following commands in gdbinit file or initGdbCommands, if used, for the debug session to properly work.
```
Comment (brainstorm) [inline docs/DEBUGGING.md:30]: @brianignacio5 Wouldn't it make sense to just have a fully defined `idf.initGdbCommands` json entry for the user to copy/paste or enable?

Comment (brainstorm) [reply docs/DEBUGGING.md:30]: Hm, nevermind, seems to be generated at the project level already?:

```
target remote :3333
symbol-file /Users/romanvg/dev/esp32s2_can/twai_network_slave/build/twai_network_slave.elf
mon reset halt
flushregs
thb app_main
c
```

File: docs/WSL.md [hunk 2]
```diff
@@ -17,6 +17,6 @@ Currently in WSL 2, we have no access to serial ports. Calling powershell.exe fr
 sudo apt-get install git wget flex bison gperf python3-pip python3-venv python3-setuptools cmake ninja-build ccache libffi-dev libssl-dev dfu-util
 ```
 
-2. Configure the extension as explained in [ONBOARDING](./ONBOARDING.md).
+2. Configure the extension as explained in [ONBOARDING](./SETUP.md).
```
Comment (pwmb) [inline docs/WSL.md:20]: Any specific reason for renaming the file, maybe we can rename the text as well?

```suggestion
2. Configure the extension as explained in [SETUP](./SETUP.md).
```

File: README.md [hunk 3]
```diff
@@ -228,23 +120,11 @@ the following:
...
-
-The search functionality allows to find a parameter by description, i.e the name that appears in the sdkconfig file.
-
-An IDF GUI Menuconfig log in Output is created to print all communications with `${idf.espIdfPath}\tools\confserver.py`. It can be be used to track any errors.
-
 ## Working with multiple projects
 
-For big projects, a user will typically have one or more projects to build, flash or monitor. The ESP-IDF uses the [Visual Studio Code Workspace file schema](https://code.visualstudio.com/docs/editor/multi-root-workspaces#_workspace-file-schema) to identify all projects folders inside the current workspace (which would be the root folder).
+For big projects, a user will typically have one or more projects to build, flash or monitor. The ESP-IDF extension follows the [Visual Studio Code Workspace file schema](https://code.visualstudio.com/docs/editor/multi-root-workspaces#_workspace-file-schema) to identify all projects folders inside the current workspace (which would be the root folder).
```
Comment (pwmb) [inline README.md:125]: Maybe move the `## Working with multiple projects` to either a new file or features file?

File: README.md [hunk 4]
```diff
@@ -265,47 +145,17 @@ Projects folders and workspace level settings are defined in the `.code-workspac
...
+You can either manually create this `.code-workspace` file and define all sub folders (projects) or when you click Menu **File** --> **Save Workspace as...** which doesn't automatically add any folder inside the current directory.
+You can add a folder to the workspace when you click Menu **File** --> **Add Folder to Workspace...**.
 
-We support connecting, viewing and editing of ESP Rainmaker enabled devices out of the box, please [refer here](/docs/ESP_RAINMAKER.md) for more info.
+> **NOTE:** You still need to manually select the debug configuration in the Debug tab that correspond to your current workspace folder. There is a project folder suffix on each debug configuration.
 
 ## Forum
 
-If you are lost at any point you can always ask question, help and suggestion in the [forum](https://spectrum.chat/espidf-vsc?tab=posts), apart from creating Github Issues. For all the [ESP-IDF](https://github.com/espressif/esp-idf) related concerns please follow [their suggested channel](https://esp32.com) of communications.
+Check out the IDEs for ESP-IDF in [ESP32 forums](https://esp32.com).
```
Comment (pwmb) [inline README.md:158]: Remove this in favour of badges (just after the title line 5)

[Forum](https://esp32.com/viewforum.php?f=40)

**Review by pwmb — CHANGES_REQUESTED** [review]
Thanks for the PR, left a few suggestions, looks very good and clean now

File: README.md [hunk 5]
```diff
@@ -12,73 +12,47 @@ Develop and debug applications for Espressif [ESP32](https://espressif.com/en/pr
...
-- Easily Build, Flash and Monitor your code for the ESP-32 and ESP32 S2 chip.
-- Syntax highlighting for [KConfig](https://docs.espressif.com/projects/esp-idf/en/latest/api-reference/kconfig.html) and ESP-IDF Kconfig style syntax validation if enabled.
-- Localization (English, Chinese, Spanish) of commands which you can also [add a language contribution](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/LANG_CONTRIBUTE.md).
-- OpenOCD server within Visual Studio Code.
-- [Code Coverage](./docs/COVERAGE.md) for editor source highlighting and generate HTML reports.
-- Search text editor's selected text in ESP-IDF documentation with **ESP-IDF: Search in documentation...** right click command or with its [keyboard shortcut](#Available-commands). Results will be shown in ESP-IDF Explorer Tab if found on ESP-IDF Documentation based on your current vscode language, ESP-IDF version in `idf.espIdfPath` (latest otherwise) and `idf.adapterTargetName`.
+The ESP-IDF extension makes it easy to develop, build, flash, monitor and debug your ESP-IDF code. See all [features](./docs/FEATURES.md).
+
```
Comment (pwmb) [inline README.md:16]: Maybe move this above the Video Preview!

**Review by pwmb — CHANGES_REQUESTED** [review]

File: docs/DEBUGGING.md (hunk 1 above)
Comment (brianignacio5) [reply docs/DEBUGGING.md:30]: If one doesn't specify initGdbCommands or gdbinitFile, the file is generated from https://github.com/espressif/esp-debug-adapter/blob/master/debug_adapter/debug_backend/debug_backend/hw_specific/esp_init.gdb replacing with the actual .elf file

Will try to describe this better in the docs.

File: README.md [hunk 6]
```diff
@@ -4,85 +4,60 @@
 
 # ESP-IDF VS Code Extension
 
+[![Forum](https://img.shields.io/badge/Forum-esp32.com-blue)](https://esp32.com/viewforum.php?f=40)
+
 Develop and debug applications for Espressif [ESP32](https://espressif.com/en/products/hardware/esp32), [ESP32-S2](https://www.espressif.com/en/products/socs/esp32-s2) chips with [ESP-IDF](https://github.com/espressif/esp-idf) IoT Development Framework.
```
Comment (pwmb) [inline README.md:9]: Also this statement feels redundant, maybe remove this and add supported chips as `badge` after `forum`

[ESP32-S2](https://espressif.com/en/products/hardware/esp32-s2)

Also in the link of this badge add another doc file with list of supported targets, boards and framework (like MDF, Arduino etc)

**Review by pwmb — CHANGES_REQUESTED** [review]
Left a few comments otherwise, LGTM

Thanks for this PR, this improves the project greatly

**Pull request merged**